Assets/BossRoom/Scripts/Client/Game/State/ClientMainMenuState.cs

```diff
@@ -9,6 +19,93 @@ namespace Unity.Multiplayer.Samples.BossRoom.Client
     /// </remarks>
     public class ClientMainMenuState : GameStateBehaviour
     {
-        public override GameState ActiveState { get { return GameState.MainMenu; } }
+        public override GameState ActiveState { get { return GameState.MainMenu;  } }
+
+        private GameNetPortal m_GameNetPortal;
+
+        private ClientGameNetPortal m_ClientNetPortal;
+
+
+        [SerializeField] private GameObject[] _autoInjected;
+        private DIScope m_Scope;
+
+        [SerializeField] private NameGenerationData m_NameGenerationData;
+        [SerializeField] private LobbyUIMediator m_lobbyUIMediator;
+
+        [SerializeField] private CanvasGroup m_mainMenuButtons;
+        [SerializeField] private GameObject m_signInSpinner;
+
+        private void Awake()
+        {
+            m_mainMenuButtons.interactable = false;
+            m_lobbyUIMediator.Hide();
+            DIScope.RootScope.InjectIn(this);
+        }
+
+        [Inject]
+        private void InjectDependenciesAndInitialize(AuthenticationAPIInterface authAPIInterface)
+        {
+            m_Scope = new DIScope(DIScope.RootScope);
+
+            m_Scope.BindInstanceAsSingle(m_NameGenerationData);
+            m_Scope.BindInstanceAsSingle(m_lobbyUIMediator);
+
+            var unityAuthenticationInitOptions = new InitializationOptions();
+
+#if UNITY_EDITOR
+            if (ClonesManager.IsClone())
+            {
+                Debug.Log("This is a clone project.");
+                var customArguments = ClonesManager.GetArgument().Split(',');
+
+                //second argument is our custom ID, but if it's not set we would just use some default.
+
+                var hardcodedProfileID = customArguments.Length > 1 ? customArguments[1] : "defaultCloneID";
+
+                unityAuthenticationInitOptions.SetProfile(hardcodedProfileID);
+            }
+#endif
+            authAPIInterface.DoSignInAsync(OnAuthSignIn,  OnSignInFailed, unityAuthenticationInitOptions);
+
+            void OnAuthSignIn()
+            {
+                m_Scope.FinalizeScopeConstruction();
+
+                foreach (var go in _autoInjected)
+                {
+                    m_Scope.InjectIn(go);
+                }
+
+                m_mainMenuButtons.interactable = true;
+                m_signInSpinner.SetActive(false);
+
+                var localUser = m_Scope.Resolve<LobbyUser>();
```

Comment:
Couldn't those be injected in the parent method as a parameter instead of having to resolve them?

Reply:
True, done.